gazelle_binary fails nogo linting

### What version of gazelle are you using?

0.36.0

### What version of rules_go are you using?

0.46.0

### What version of Bazel are you using?

7.1.1

### Does this issue reproduce with the latest releases of all the above?

Yes (unless there's been a release in the last 24 hours or so)

### What operating system and processor architecture are you using?

OSX 14.4.1 arm64

### What did you do?

- Define a `gazelle_binary` rule (with any languages -- `DEFAULT_LANGUAGES` suffices)
- enable nogo
- build the gazelle binary target

See https://github.com/sacsar/bazel-go-repro/tree/main

### What did you expect to see?

A successful build.

### What did you see instead?
```
❯ bazel build //:my_gazelle
Starting local Bazel server and connecting to it...
INFO: Analyzed target //:my_gazelle (164 packages loaded, 15502 targets configured).
ERROR: /Users/scsar/external_repos/bazel_go_repro/BUILD.bazel:10:15: GoCompilePkg my_gazelle.a failed: (Exit 1): builder failed: error executing GoCompilePkg command (from target //:my_gazelle) bazel-out/darwin_arm64-opt-exec-ST-13d3ddad9198/bin/external/rules_go~~go_sdk~main__download_0/builder_reset/builder compilepkg -sdk external/rules_go~~go_sdk~main__download_0 -installsuffix ... (remaining 73 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
compilepkg: nogo: errors found by nogo during build-time code analysis:
external/gazelle~/cmd/gazelle/fix-update.go:127:8: declaration of "err" shadows declaration at line 111 (shadow)
external/gazelle~/cmd/gazelle/fix-update.go:291:5: declaration of "err" shadows declaration at line 256 (shadow)
external/gazelle~/cmd/gazelle/fix-update.go:307:6: declaration of "err" shadows declaration at line 256 (shadow)
external/gazelle~/cmd/gazelle/fix-update.go:373:10: declaration of "err" shadows declaration at line 256 (shadow)
Target //:my_gazelle failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 7.484s, Critical Path: 0.82s
INFO: 2 processes: 2 internal.
ERROR: Build did NOT complete successfully
```

Discussion on Slack: https://bazelbuild.slack.com/archives/CDBP88Z0D/p1712365708985129 @fmeum suggests that in addition to getting the linting to pass, it might be worth exploring whether `gazelle_binary` could precompile the Gazelle sources within the gazelle repo (meaning nogo would skip them).

--- a/cmd/gazelle/fix-update.go
+++ b/cmd/gazelle/fix-update.go
@@ -28,6 +28,8 @@ import (
 	"strings"
 	"syscall"
 
+	"github.com/bazelbuild/buildtools/build"
+
 	"github.com/bazelbuild/bazel-gazelle/config"
 	gzflag "github.com/bazelbuild/bazel-gazelle/flag"
 	"github.com/bazelbuild/bazel-gazelle/internal/wspace"
@@ -38,7 +40,6 @@ import (
 	"github.com/bazelbuild/bazel-gazelle/resolve"
 	"github.com/bazelbuild/bazel-gazelle/rule"
 	"github.com/bazelbuild/bazel-gazelle/walk"
-	"github.com/bazelbuild/buildtools/build"
 )
 
 // updateConfig holds configuration information needed to run the fix and
@@ -125,7 +126,7 @@ func (ucr *updateConfigurer) CheckFlags(fs *flag.FlagSet, c *config.Config) erro
 		if !filepath.IsAbs(dir) {
 			dir = filepath.Join(c.WorkDir, dir)
 		}
-		dir, err := filepath.EvalSymlinks(dir)
+		dir, err = filepath.EvalSymlinks(dir)
 		if err != nil {
 			return fmt.Errorf("%s: failed to resolve symlinks: %v", arg, err)
 		}
@@ -220,7 +221,7 @@ func (ucr *updateConfigurer) KnownDirectives() []string { return nil }
 
 func (ucr *updateConfigurer) Configure(c *config.Config, rel string, f *rule.File) {}
 
-// visitRecord stores information about about a directory visited with
+// visitRecord stores information about a directory visited with
 // packages.Walk.
 type visitRecord struct {
 	// pkgRel is the slash-separated path to the visited directory, relative to
@@ -289,7 +290,7 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 	}
 	ruleIndex := resolve.NewRuleIndex(mrslv.Resolver, exts...)
 
-	if err := fixRepoFiles(c, loads); err != nil {
+	if err = fixRepoFiles(c, loads); err != nil {
 		return err
 	}
 
@@ -372,7 +373,8 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 		}
 
 		maybeRecordReplacement := func(ruleKind string) (*string, error) {
-			repl, err := lookupMapKindReplacement(c.KindMap, ruleKind)
+			var repl *config.MappedKind
+			repl, err = lookupMapKindReplacement(c.KindMap, ruleKind)
 			if err != nil {
 				return nil, err
 			}
@@ -463,8 +465,8 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 
 	if len(errorsFromWalk) > 1 {
 		var additionalErrors []string
-		for _, error := range errorsFromWalk[1:] {
-			additionalErrors = append(additionalErrors, error.Error())
+		for _, err = range errorsFromWalk[1:] {
+			additionalErrors = append(additionalErrors, err.Error())
 		}
 
 		return fmt.Errorf("encountered multiple errors: %w, %v", errorsFromWalk[0], strings.Join(additionalErrors, ", "))
@@ -480,7 +482,7 @@ func runFixUpdate(wd string, cmd command, args []string) (err error) {
 			err = cerr
 		}
 	}()
-	if err := maybePopulateRemoteCacheFromGoMod(c, rc); err != nil {
+	if err = maybePopulateRemoteCacheFromGoMod(c, rc); err != nil {
 		log.Print(err)
 	}
 	for _, v := range visits {

